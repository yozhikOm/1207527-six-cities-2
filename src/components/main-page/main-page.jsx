import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {Header} from '../header/header.jsx';
import {Menu} from '../menu/menu.jsx';
import {Properties} from '../properties/properties.jsx';
import withActiveItem from '../../hocs/with-active-item/with-active-item.js';

const PropertiesWrapped = withActiveItem(Properties);

class MainPage extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      currentCity,
      cities,
      offers,
      onCityClick,
    } = this.props;

    return (
      <div className="page page--gray page--main">
        <Header />
        <main className="page__main page__main--index">
          <Menu cities={cities} onCityClick={onCityClick}/>

          <PropertiesWrapped offers={offers} currentCity={currentCity}/>

        </main>
      </div>
    );
  }
}

MainPage.propTypes = {
  currentCity: PropTypes.shape({
    title: PropTypes.string.isRequired,
    coordinates: PropTypes.arrayOf(PropTypes.number, PropTypes.number).isRequired,
  }),
  cities: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        coordinates: PropTypes.arrayOf(PropTypes.number, PropTypes.number).isRequired,
      })
  ),
  offers: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        location: PropTypes.shape({
          city: PropTypes.string.isRequired,
          coordinates: PropTypes.arrayOf(PropTypes.number, PropTypes.number).isRequired,
        }).isRequired,
        title: PropTypes.string.isRequired,
        type: PropTypes.string.isRequired,
        price: PropTypes.number.isRequired,
        description: PropTypes.string.isRequired,
        photos: PropTypes.arrayOf(
            PropTypes.shape({
              src: PropTypes.string,
            })
        ),
        host: PropTypes.string.isRequired,
      })
  ),
  onCityClick: PropTypes.func.isRequired,
};

export {MainPage};
