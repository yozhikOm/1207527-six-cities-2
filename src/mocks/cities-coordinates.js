export const citiesCoordinates = [
  {
    "title": `Amsterdam`,
    "coordinates": [52.38333, 4.9],
  },
  {
    "title": `Berlin`,
    "coordinates": [52.5200, 13.4050],
  },
  {
    "title": `New York`,
    "coordinates": [40.7128, -74.0060],
  },
];
