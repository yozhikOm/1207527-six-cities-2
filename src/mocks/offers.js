export const offers = [
  {
    "id": 1,
    "title": `Cozy house amsterdam`,
    "type": `House`,
    "price": 650,
    "photos": [
      {
        "src": `img/apartment-01.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.3909553943508, 4.85309666406198],
  },
  {
    "id": 2,
    "title": `Cosy room in spacious apartment`,
    "type": `Room`,
    "price": 45,
    "photos": [
      {
        "src": `img/apartment-02.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.369553943508, 4.85309666406198],
  },
  {
    "id": 3,
    "title": `Elegant private room in stylish house`,
    "type": `Room`,
    "price": 128,
    "photos": [
      {
        "src": `img/room.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.3909553943508, 4.929309666406198],
  },
  {
    "id": 4,
    "title": `Huge Tinyhouse in the green, between sea and city`,
    "type": `House`,
    "price": 1520,
    "photos": [
      {
        "src": `img/apartment-03.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.3809553943508, 4.939309666406198],
  }
];
