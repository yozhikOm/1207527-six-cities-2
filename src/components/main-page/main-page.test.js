import React from 'react';
import {MainPage} from './main-page.jsx';
import ShallowRenderer from 'react-test-renderer/shallow';
import {offers} from '../../mocks/offers.js';
import {citiesCoordinates} from '../../mocks/cities-coordinates.js';

it(`MainPage компонент рендерится корректно`, () => {
  const renderer = new ShallowRenderer();
  const mockCurrCity = {
    title: `Amsterdam`,
    coordinates: [0, 0],
  };

  renderer.render(<MainPage
    currentCity={mockCurrCity}
    cities={citiesCoordinates}
    offers={offers}
    onCityClick={jest.fn()}
  />);

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
