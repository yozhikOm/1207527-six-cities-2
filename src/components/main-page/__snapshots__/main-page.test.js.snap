// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage компонент рендерится корректно 1`] = `
<div
  className="page page--gray page--main"
>
  <Header />
  <main
    className="page__main page__main--index"
  >
    <Menu />
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            312 places to stay in Amsterdam
          </b>
          <Sorting />
          <Properties
            items={
              Array [
                Object {
                  "coordinates": Array [
                    52.3909553943508,
                    4.85309666406198,
                  ],
                  "description": "In onze eigen achtertuin hebben we de oude sauna omgebouwd tot gastenverblijf met eigen badkamer en toilet. Er is geen kook mogelijkheid, maar wel een koelkast en wastafel. Wij wonen in een rustige buurt welke goed bereikbaar is met het openbaar vervoer en/of eigen vervoer. Grote steden als haarlem, amsterdam en alkmaar zijn vlakbij. Het strand is ca. 10 minuten met de auto. Houd u van rust en wilt u toch dichtbij de grote steden zijn dan is dit de ideale plek om te verblijven!",
                  "host": "Maria",
                  "id": 1,
                  "photos": Array [
                    Object {
                      "src": "img/apartment-01.jpg",
                    },
                    Object {
                      "src": "img/amsterdam.jpg",
                    },
                  ],
                  "price": 650,
                  "title": "Cozy house amsterdam",
                  "type": "House",
                },
                Object {
                  "coordinates": Array [
                    52.369553943508,
                    4.85309666406198,
                  ],
                  "description": "Sparkling clean, comfy and private room close to the city center. The place is super sweet with coffee & tea available.",
                  "host": "Giovanni",
                  "id": 2,
                  "photos": Array [
                    Object {
                      "src": "img/apartment-02.jpg",
                    },
                    Object {
                      "src": "img/room-2.jpg",
                    },
                    Object {
                      "src": "img/amsterdam.jpg",
                    },
                  ],
                  "price": 45,
                  "title": "Cosy room in spacious apartment",
                  "type": "Room",
                },
                Object {
                  "coordinates": Array [
                    52.3909553943508,
                    4.929309666406198,
                  ],
                  "description": "From this penthouse room guests have a breathtaking view of the city. When you wake up Amsterdam is happening all around you. We guarantee you maximum sleeping comfort with our quality Tempur mattresses.",
                  "host": "Tio Leo",
                  "id": 3,
                  "photos": Array [
                    Object {
                      "src": "img/room.jpg",
                    },
                    Object {
                      "src": "img/amsterdam.jpg",
                    },
                  ],
                  "price": 128,
                  "title": "Elegant private room in stylish house",
                  "type": "Room",
                },
                Object {
                  "coordinates": Array [
                    52.3809553943508,
                    4.939309666406198,
                  ],
                  "description": "This compact guesthouse/design studio is very suitable for both working and relaxing. Plenty of working space, kitchenette with microwave and fridge, a cosy fire place (gas) and upstairs a large lounge bed. The 'in between' sized lounge bed (120 cm wide) is very comfortable for one person, but it could also sleep a couple. The stairs in this guesthouse studio are pretty steep, but safe. Comfortable floor heating downstairs.",
                  "host": "Rebekka",
                  "id": 4,
                  "photos": Array [
                    Object {
                      "src": "img/apartment-03.jpg",
                    },
                    Object {
                      "src": "img/amsterdam.jpg",
                    },
                  ],
                  "price": 1520,
                  "title": "Huge Tinyhouse in the green, between sea and city",
                  "type": "House",
                },
              ]
            }
          />
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <Map
              coordinatesArray={
                Array [
                  Array [
                    52.3909553943508,
                    4.85309666406198,
                  ],
                  Array [
                    52.369553943508,
                    4.85309666406198,
                  ],
                  Array [
                    52.3909553943508,
                    4.929309666406198,
                  ],
                  Array [
                    52.3809553943508,
                    4.939309666406198,
                  ],
                ]
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
